testing/simapp/app.go

```diff
@@ -332,22 +334,24 @@ func NewSimApp(
 	)
 	transferModule := transfer.NewAppModule(app.TransferKeeper)
 
+	// NOTE: the IBC mock keeper and application module is used only for testing core IBC. Do
+	// note replicate if you do not need to test core IBC or light clients.
+	mockModule := ibcmock.NewAppModule(scopedIBCMockKeeper, &app.IBCKeeper.PortKeeper)
+	icaMockModule := ibcmock.NewAppModule(scopedICAMockKeeper, &app.IBCKeeper.PortKeeper)
+
 	app.ICAKeeper = icakeeper.NewKeeper(
 		appCodec, keys[icatypes.StoreKey],
 		app.IBCKeeper.ChannelKeeper, &app.IBCKeeper.PortKeeper,
 		app.AccountKeeper, scopedICAKeeper, app.MsgServiceRouter(),
 	)
 	icaModule := ica.NewAppModule(app.ICAKeeper)
-	icaIBCModule := ica.NewIBCModule(app.ICAKeeper, nil)
-
-	// NOTE: the IBC mock keeper and application module is used only for testing core IBC. Do
-	// note replicate if you do not need to test core IBC or light clients.
-	mockModule := ibcmock.NewAppModule(scopedIBCMockKeeper, &app.IBCKeeper.PortKeeper)
+	icaIBCModule := ica.NewIBCModule(app.ICAKeeper, icaMockModule)
 
```

Comment:
Just thinking out loud but is there any quick way to initiate a mock ica-auth (base) module here so we can ensure the integration works with the middleware?

Reply:
I could give it a shot, we would need to add functions to ica-auth

Reply:
ibc_module_test does testing of the ica auth module callbacks. What it does not do is test the flow of calling `TrySendTx`, but I don't really see this as necessary since we already test claiming a capability with an ica auth module correctly routes to the middleware and then the auth module

Reply:
Yeah, that's a good point. I didn't see `ibc_module_test` on my first review, github collapsed the file so I missed it. Tests look great :+1: 

I think it might be nice to have the `onRecv/timeout/ack` tests which essentially test the previous hook functionality but we can follow up with another PR :handshake: